_drafts/shallow-clone-post-mortem.md

```diff
@@ -0,0 +1,58 @@
+[Recently](https://github.com/CocoaPods/CocoaPods/issues/4989) we were made aware of issues that our users were having
+cloning the [CocoaPods Master spec-repo](https://github.com/CocoaPods/Specs), which contains specifications for projects
+shared _by_ and _with_ the community.
+[The GitHub infrastructure team](https://github.com/CocoaPods/CocoaPods/issues/4989#issuecomment-193772935) were quick
+to explain that this was because of us hitting CPU rate limits on their end, which turned out to be caused, besides the
+high volume of clones, by the default options we chose to clone this repo.
+
+After some back and forth, we settled on the following list of causes and solutions:
+
+1. To ease initial setup for CocoaPods users, we performed shallow clones of spec-repos, meaning they only had to fetch
+   the latest state of the repo. However, this meant that the GitHub servers had to spend much more time figuring out
+   which Git objects the user already has, needs, and compute the deltas for those objects.
+
+   Therefore, we are no longer going to shallow clone spec-repos on initial setup and we’ll automatically convert
+   existing shallow clones to full clones.
+
+2. To ensure the user always has access to all the available podspecs, we automatically performed a `git fetch` on each
+   `pod install` and `pod update`. The amount of fetches this resulted in made the above shallow clone issue even worse.
+   That, combined with the fact that on most `pod install` runs there was little need to actually update the repo, led
+   to us changing the following:
+
+   - We will no longer automatically update spec-repos on `pod install`. If a lockfile exists and podspecs are missing,
+     an error will be raised that suggests the user to update their spec-repos with `pod repo update`.
+
+   - We will still automatically update the spec-repos on `pod update`, because that’s implicitly what the user asks for
+     at that time, but it’s an action taken far less often than running `pod install`.
+
+   - Before _any_ spec-repo fetching, we’ll be making use of a GitHub API that returns if a repo has any changes in a
```

Comment:
Might be worth clarifying that this only applies to the master specs repo at the moment?